apps/prairielearn/src/lib/manualGrading.sql

```diff
@@ -377,7 +377,10 @@ SELECT
   iq.manual_points,
   s.manual_rubric_grading_id,
   $check_modified_at::TIMESTAMPTZ IS NOT NULL
-  AND $check_modified_at != iq.modified_at AS modified_at_conflict
+  -- We are comparing a database timestamp (microseconds precision) to a time
+  -- that comes from the client (milliseconds precision). This avoids a precision
+  -- mismatch that would cause the comparison to fail.
+  AND date_trunc('milliseconds', $check_modified_at) != date_trunc('milliseconds', iq.modified_at) AS modified_at_conflict
```

Comment:
Just to confirm: is this operation equivalent to the truncation/rounding done when the database value is retrieved? Or asking the opposite: is it possible that some process that converts the value from Postgres into JS/TS/ISO rounds the value so that this operation would not match?

To be on the safe side: would a better alternative be to compare the two so that the difference is less than 1 millisecond? Something like:
```suggestion
  AND NOT ($check_modified_at::timestamptz - iq.modified_at) BETWEEN '-1 ms' AND '1 ms' AS modified_at_conflict
```

If you have reason to believe that the truncation is safe, I'm ok with the current implementation, just trying to cover all bases.

Reply:
It sounds like what you are asking is if a date with microseconds in javascript is ever rounded. I don't think it is. The value would get coverted with this:

```ts
export const DateFromISOString = z
  .union([z.string(), z.date()])
  .refine(
    (s) => {
      const date = new Date(s);
      return !Number.isNaN(date.getTime());
    },
    {
      message: 'must be a valid ISO date string',
    },
  )
  .transform((s) => new Date(s));
```

<img width="822" height="186" alt="CleanShot 2025-07-11 at 17 58 08@2x" src="https://github.com/user-attachments/assets/65aee70c-a6da-4cb0-84b3-1b222dc080fa" />

Reply:
Sounds good, I'm convinced.